packages/blade/src/components/Toast/_decisions/decisions.md

```diff
@@ -0,0 +1,251 @@
+
+## Simple Usage
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const HomePage = () => {
+  const { showToast } = useToast();
+
+  return (
+    <Button
+      onClick={() => {
+        showToast({
+          type: 'informational',
+          color: 'success',
+          content: 'Payment Successful',
+          leading: <DollarIcon />,
+          autoDismiss: true,
+          onDismissButtonClick: () => {
+            console.log('Toast dismissed');
+          },
+          action: {
+            text: 'View',
+            onClick: () => {
+              console.log('Toast action clicked');
+            }
+          }
+        });
+      }}
+    >
+      Show Toast
+    </Button>
+  );
+};
+
+const App = () => {
+  return (
+    <BladeProvider>
+      <ToastContainer position="bottom-left" />
+      <HomePage />
+    </BladeProvider>
+  )
+}
+```
+
+## Props
+
+### ToastContainer
+
+```ts
+type ToastContainer = {
+  position: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+}
+```
+
+```ts
+type Toast = {
+  /**
+   * @default `informational`
+   */
+  type: 'informational' | 'promotional';
+
+  /**
+   * @default `neutral`
+   */
+  color: 'neutral' | 'positive' | 'negative' | 'warning' | 'information'
+
+  /**
+   * If the type is `promotional`, the content will be `React.ReactNode`
+   */
+  content: string | React.ReactNode;
+
+  /**
+   * Can be used to render an icon
+   */
+  leading?: IconComponent;
+
+  /**
+   * If true, the toast will be dismissed after few seconds
+   * 
+   * Duration for promotional toast is 6s
+   * Duration for informational toast is 4s
+   * 
+   * @default false
+   */
+  autoDismiss?: boolean;
+
+  /**
+   * Called when the toast is dismissed or duration runs out
+   */
+  onDismissButtonClick?: () => void;
+
+  /**
+   * Primary action of toast
+   */
+  action?: {
+    text: string;
+    onClick?: () => void;
+    isLoading? boolean;
+  }
+
+  /**
+   * Forwarded to react-hot-toast
+   * 
+   * This can be used to programatically update toasts by providing an id
+   */
+  id?: string; 
+}
+```
+
+### useToast
+
+The useToast hook will return few modified methods from [react-hot-toast](https://react-hot-toast.com/docs/toast): 
+
+```ts
+type useToastReturnType = {
+  /**
+   * @returns id of the toast
+   */
+  showToast: (toast: Toast) => string;
+
+  /**
+   * id of the toast to be dismissed
+   * 
+   * if id is not provided, all the toasts will be dismissed
+   */
+  dismissToast: (id?: string) => void;
+}
+```
+
+## Examples
+
+#### Removing a toast
+
+react-hot-toast provides this functionality, for more info see [react-hot-toast docs](https://react-hot-toast.com/docs/toast#dismiss-toast-programmatically)
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const Example = () => {
+  const toastId = React.useRef(null);
+  const { showToast, dismissToast } = useToast();
```

Comment:
toast will have a context at the app level right? can't see that in the example?

Reply:
Maybe, I'm not sure right now actually, so react-hot-toast works in a way that it doesn't need ReactContext. 
In the current POC that I've done that also didn't need a toast context.